File: leetcode/binary/leetcode_191.java
package leetcode.binary;

public class leetcode_191 {
    public static void main(String[] args) {
        leetcode_191 lc = new leetcode_191();
        System.out.println(lc.hammingWeight(00000000000000000000000000001011));
    }

    public int hammingWeight(int n) {
        int count = 0;

        while (n != 0) {
            n = n & (n - 1);
            count++;
        }

        return count;
    }
}

File: bit_manipulation/LeetCode_191.kt
package bit_manipulation

class LeetCode_191 {

    fun hammingWeight(n: Int) = n.toUInt().countOneBits()

}

fun main() {
    // Input
    val n = 11
    // Output
    LeetCode_191().hammingWeight(n).run {
        println(this)
        require(3 == this)
    }
}
